docs/0.3-DRAFT/index.md

```diff
@@ -1378,7 +1328,90 @@ To record curation actions which modify a [File](http://schema.org/MediaObject)
+
+The distinction is fluid, and comes down to the availability and understandability of code to re-run a workflow. For instance, office spreadsheet applications are generally available and do not need further explanation; while a Python script that is customized for a particular data analysis might be important to understand further and SHOULD be included as source code in the RO-Crate dataset:
+
+```json
+       {
+            "@id": "workflow/retropath.knime",  
+            "@type": ["File", "SoftwareSourceCode", "Workflow"],
+            "name": "RetroPath Knime workflow",
+            "description": "KNIME implementation of RetroPath2.0 workflow",
+            "author": {"@id": "https://orcid.org/0000-0001-2345-6789"},
+            "programmingLanguage": {"@id": "https://www.knime.com/knime-software/knime-analytics-platform"},
+            "license": "https://spdx.org/licenses/BSD-2-Clause.html",
+            "potentialAction": {
+                "@type": "ActivateAction",
+                "instrument": {"@id": "https://www.knime.com/knime-software/knime-analytics-platform"}
+            }
+        }
+```
+
+Workflows and scripts are saved on disk using a language and generally need a runtime, in RO-Crate this SHOULD be indicated using a liberal interpretation of `programmingLanguage` and `potentialAction`. Note that the language and its runtime might differ, e.g. there are multiple potential runtimes of a JavaScript, although in the base-case they are the same, and as a shortcut can be described in one go as both a `ComputerLanguage` and `SoftwareApplication`:
+
+```json
+       {
+            "@id": "https://www.knime.com/knime-software/knime-analytics-platform",
+            "@type": [
+                "ComputerLanguage",
+                "SoftwareApplication"
+            ],
+            "name": "KNIME Analytics Platform",
+            "alternateName": "KNIME",
+            "url": {"@id": "https://www.knime.com/knime-software/knime-analytics-platform"},
+            "version": "3.6"
+        }
+```
+
+A data entity representing a runtime application or language MUST have a [name], [url] and [version], which SHOULD indicate a known version the workflow/script was developed or tested with. [alternateName] MAY be provided if there is a shorter colloquial name, for instance “R” instead of “The R Project for Statistical Computing”.
+
+To indicate that a `SoftwareSourceCode` is in the form of a script use a `@type` of `WorkflowScript`. If the `SoftwareSourceCode` is in the form of a workflow (e.g. a pipeline of steps with data flow) `Workflow`.
+
+#### Workflow diagram/sketch
+
+It is useful to show a diagram or sketch to explain the script/workflow. This may have been generated from the workflow management system or drawn manually as a diagram. It SHOULD be included as an `ImageObject` which is `about` the `SoftwareSourceCode`:
```

Comment:
What do you think, should we keep this as `about` from the image to the workflow, or [`image`](http://schema.org/image) **from** the workflow to the image?  The image may not have been generated programmatically from the workflow file, but often is.